games-strategy/freeciv/freeciv-3.1.0_beta2.ebuild

```diff
@@ -0,0 +1,193 @@
+	KEYWORDS="~amd64 ~ppc64 ~x86"
+	MY_P="${PN}-${MY_PV}"
+	S="${WORKDIR}/${MY_P}"
+fi
+
+LICENSE="GPL-2+"
+SLOT="0"
+IUSE="dedicated gtk3 gtk4 json mapimg modpack mysql nls +qt5 qt6 readline rule-editor sdl +sound +system-lua web-server "
+
+# I'm pretty sure that you can't build both qt flavours at the same time
+REQUIRED_USE="
+	system-lua? ( ${LUA_REQUIRED_USE} )
+	!dedicated? ( || ( gtk3 gtk4 qt5 qt6 sdl ) )
+	qt5?  ( !qt6 )
+	qt6?  ( !qt5 )
+"
+
+RDEPEND="
+	app-arch/bzip2
+	app-arch/xz-utils
+	app-arch/zstd:=
+	dev-db/sqlite:3
+	dev-libs/icu:=
+	net-misc/curl
+	sys-devel/libtool
+	sys-libs/zlib
+	!dedicated? (
+		media-libs/libpng
+		gtk3? ( x11-libs/gtk+:3 )
+		gtk4? ( gui-libs/gtk:4 )
+		mapimg? ( media-gfx/imagemagick:= )
+		nls? ( virtual/libintl )
+		qt5? (
+			dev-qt/qtcore:5
+			dev-qt/qtgui:5
+			dev-qt/qtwidgets:5
+		)
+		qt6? (
+			dev-qt/qtbase:6[gui,widgets]
+		)
+		sdl? (
+			media-libs/libsdl2[video]
+			media-libs/sdl2-gfx
+			media-libs/sdl2-image[png]
+			media-libs/sdl2-ttf
+		)
+		sound? (
+			media-libs/libsdl2[sound]
+			media-libs/sdl2-mixer[vorbis]
+		)
+	)
+	json? ( dev-libs/jansson:= )
+	readline? ( sys-libs/readline:= )
+	system-lua? (
+		${LUA_DEPS}
+	)
+"
+DEPEND="${RDEPEND}
+	!dedicated? ( x11-base/xorg-proto )
+"
+# Calls gzip during build
+BDEPEND="
+	app-arch/gzip
+	virtual/pkgconfig
+	nls? ( sys-devel/gettext )
+"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-lua-search.patch
+)
+
+pkg_setup() {
+	use system-lua && lua-single_pkg_setup
+}
+
+src_configure() {
+	local myclient=() emesonargs=() myfcmp=()
+
+	# Upstream considers meson "experimental" until 3.2.0 according to their roadmap
+	emesonargs+=( -Dack_experimental=true )
+
+	# meson build drops the ability to _not_ build a server in favour
+	# of optionally replacing the server binary the freeciv-web backend
+	emesonargs+=(
+		$(meson_use web-server freeciv-web)
+	)
+
+	# Select any client backends that we want enabled; dedicated server shouldn't build a UI
+	# for sanity we'll build the modpack bin with the same UIs as the client.
+	# 'fcmp' = freeciv modpack (client) - gtk3, qt, cli, gtk4
+	if ! use dedicated ; then
+		# there's no SDL modpack backend; rather than incidentally pull in GTK3 (as is default)
+		# let's explicitly set the backend to CLI
+		use sdl && myclient+=( sdl2 ) && use modpack && myfcmp+=( cli )
+		use gtk3 && myclient+=( gtk3.22 ) && use modpack && myfcmp+=( gtk3 )
+		use gtk4 && myclient+=( gtk4 ) && use modpack && myfcmp+=( gtk4 )
+		if use qt5 ; then
+			emesonargs+=( -Dqtver=qt5 )
+			myclient+=( qt )
+			use modpack && myfcmp+=( qt )
+		fi
+		if use qt6 ; then
+			emesonargs+=( -Dqtver=qt6 )
+			myclient+=( qt )
+			use modpack && myfcmp+=( qt )
+		fi
+	else
+		if use modpack ; then
+			myfcmp+=( cli )
+		fi
+	fi
+
+	# the client and fpmc arrays are now populated (or not for dedicated); let's add them to+ emesonargs
+emesonargs+=(
+		-Dclients=$(echo ${myclient[*]} | sed 's/ /,/g')
```

Comment:
Can we do:
-Dclients="${myclient[@]/#/,}"
or so?

Reply:
not trivially; that results in `,client1 ,client2 ,...` and makes meson very unhappy.